### DISCUSSION TITLE
Post method

### DISCUSSION THREAD

**Discussion by guthux** [mockoon/mockoon#562] [Help]
Hello everyone!

Can anybody tell me if it is possible to generate a non Get method?

I can't find the way to send a request with body.

**Obi-Wan Kenobi help me, You're my only hope.**

**Comment by 255kb** [comment]
You can create a POST response using the method menu:
![create-api-route.gif](https://user-images.githubusercontent.com/7489814/132910481-e9b38658-3a9d-4841-99a4-856c8ae05027.gif)
Please note that Mockoon accepts and parse entering bodies on any method, which means that you can use [rules](https://mockoon.com/docs/latest/route-responses/dynamic-rules/) or [templating helpers](https://mockoon.com/docs/latest/templating/overview/) on GET routes too.

May the API mock be with you.

**Comment by guthux** [comment]
It is possible to create a post method but it won't work since you need to send a body for the request and there is no field available for that.
Thanks in advance.

**Reply by 255kb** [reply]
I am not sure to understand what you mean. There is a body editor where you can provide the body content with which Mockoon should respond. And you can send bodies to Mockoon that will get parsed no matter the method used.

Could you please elaborate on what exactly you are trying to achieve and how? And what is not working? Thank you 🙂

**Reply by guthux** [reply]
I'm trying to mock services with Mockoon so I can used them from IntelliJ. I'm learning automation for back ends.

The thing is that I need one field "Body" in a get method. 
But for a post method I need two fields, one for the request and one for the response. But I have only one.

**Reply by 255kb** [reply]
Which fields are you talking about? If they are part of Mockoon's UI could you please post screen captures? Thank you.

**Reply by guthux** [reply]
"There is a body editor where you can provide the body content with which Mockoon should respond. "
Exactly the body editor is for the response.

"And you can send bodies to Mockoon that will get parsed no matter the method used."
How can I send those bodies? 

Can you give me a post example please?

**Reply by 255kb** [reply]
Mockoon is an API mocking tool, so indeed it only care about what to respond when called.
To make a call to a running mock you must either use curl, postman, insomnia, or similar tools.
The most straightforward is to run a curl command in your terminal:
`curl -d '{"key1":"value1", "key2":"value2"}' -H "Content-Type: application/json" -X POST http://localhost:3000/data`
You must update `http://localhost:3000/data` with the correct port where the mock is running (3000 is the default) and the route path you set up in Mockoon.

I advise you to use a tool like postman to make requests to API in general.
You will find many tutorials online.

**Reply by AndreiPashkin** [reply]
@255kb 

Is it possible to somehow specify not only URL but request body so that they could be exported as OpenAPI specification?

**Reply by 255kb** [reply]
@AndreiPashkin Mockoon is not an API design tool, at least not yet. So, most request's info cannot be provided.

**Comment by guthux** [comment]
Please help me, I really need to know how to use this software for my company.
It could become a standard for us.

**Comment by sharu725** [comment]
@guthux you can do it. 
The first **Body** field is the response you get from the POST request. 

But to send something along with the POST request, you can use **Rules**
![Screenshot from 2022-02-20 17-46-57](https://user-images.githubusercontent.com/8033084/154842140-a776d9d4-8d21-43c9-a358-66a4b5d6c160.png)

**Reply by adurage** [reply]
Hadley found such an example when I want to match my request body for post request. is this the only way to do this?

**Comment by myatthitlwin8** [comment]
Hello everyone!

Can anybody help me with this case?
I created POST method endpoint with rules like the below ScreenShots :
It works properly in my local when calling with postman but didn't work after deploying in Linux server and only get the default response to whatever request changes. Why does it happen? Other GET method rules work properly in both local and server
![Postman200](https://user-images.githubusercontent.com/91549615/209911802-1421db89-ee17-4091-be84-16e90fcf1d23.PNG)
![Postman404](https://user-images.githubusercontent.com/91549615/209911807-cd8bf116-e99b-4f36-9775-0343324ddfc0.PNG)
![Response200](https://user-images.githubusercontent.com/91549615/209911812-1ab39fec-d262-4bc4-b526-8f1909c3db02.PNG)
![Response404](https://user-images.githubusercontent.com/91549615/209911818-75f251bc-7f07-4827-9dc7-5b1b3cb56e96.PNG)
![Rule200](https://user-images.githubusercontent.com/91549615/209911820-a710df9c-1510-40f4-bb7c-b127006e2b89.PNG)
.Please......

**Reply by 255kb** [reply]
I see nothing wrong with your setup. If it works locally it should work using the CLI or on another computer. 
I suspect that this is a firewall issue. Frequently, default firewall settings block requests other than GET, especially the ["F5" firewall](https://clouddocs.f5.com/training/community/waf/html/waf341/module2/lab1/lab1.html).

**Reply by myatthitlwin8** [reply]
**Thanks for helping me accomplish my goal.** @255kb
